lib/Doctrine/Connection/Statement.php

```diff
@@ -261,13 +269,19 @@ public function execute($params = null)
 
             $this->_conn->getListener()->postStmtExecute($event);
 
-            return $result;
         } catch (PDOException $e) {
+            $result = false;
+            $this->_conn->rethrowException($e, $this);
         } catch (Doctrine_Adapter_Exception $e) {
+            $result = false;
+            $this->_conn->rethrowException($e, $this);
+        } finally {
```

Comment:
`finally` was only introduced in PHP 5.5 and the lib should stay compatible with at least PHP 5.3